src/main/java/net/sf/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -339,6 +340,44 @@ private BracketChecker(String field) {
         }
     }
 
+    private static class EditionChecker implements Checker {
+
+        private static final Predicate<String> FIRST_LETTER_CAPITALIZED = Pattern.compile("^[A-Z]").asPredicate();
+        private static final Predicate<String> ONLY_NUMERALS_OR_LITERALS = Pattern.compile("^([0-9]+|[^0-9]+)$")
+                .asPredicate();
+
+
+        /**
+         * Checks, if the first letter is capitalized (BibTeX mode)
+         * Checks, if field contains only an integer (BibLaTeX mode)
+         * Official bibtex spec:
+         * The edition of a book-for example, “Second”.
+         * This should be an ordinal, and should have the first letter capitalized.
+         * BibLaTex:
+         * The edition of a printed publication. This must be an integer, not an ordinal.
+         * It is also possible to give the edition as a literal string, for example "Third, revised and expanded edition".
+         */
+        @Override
+        public List<IntegrityMessage> check(BibEntry entry) {
+            Optional<String> value = entry.getField(FieldName.EDITION);
+            if (!value.isPresent()) {
+                return Collections.emptyList();
+            }
+            //BibTeX
+            if (!FIRST_LETTER_CAPITALIZED.test(value.get().trim()) && (!bibDatabaseContext.isBiblatexMode())) {
```

Comment:
I would have changed the order of the statements in the conditions, checking for bibllatex mode first: